### PR TITLE
Write tests with jest (Coverage 100%)

### LINKED ISSUE #1: Add tests with Jest

### PR DESCRIPTION
**Pull request by timhagn**
As promised, here the tests written in jest.
Created a second pull request, so that you have it easier to test your workflow with the
updated dependencies.

**For the tests to work I did the following:**
- [x] all the steps from my other PR #14
- [x] cleaned up package.json and removed unneeded entries
- [x] added jest (^3.6.0), jest-styled-components (^6.2.2), react (^16.6.0), react-dom (^16.6.0),
       react-test-renderer (^16.6.0), @babel/preset-react (^7.0.0), babel-jest (^23.6.0) & 
       babel-core (^7.0.0-bridge.0, needed for jest to work flawlessly with Babel 7) as devDependencies
- [x] wrote tests for convertors.js (Line, Function & Statement Coverage @ 100%, 
       Branch Coverage @ 80%)
- [x] wrote tests for index.js (Line, Function & Statement Coverage @ 100%, 
       Branch Coverage @ 83.33%)

```
=============================== Coverage summary ===============================
Statements   : 100% ( 36/36 )
Branches     : 81.82% ( 9/11 )
Functions    : 100% ( 14/14 )
Lines        : 100% ( 30/30 )
================================================================================
```

**Things that might be further done:**
- bump Branch coverage to 100% (but as some of the branches jest complains about are private
      I have no clue at the moment, how this could be done)
- Add test report to README.md to "brag about it" ^^

**"Breaking" changes:**
- Adding react & react-dom at their latest version is necessary for the tests - but as they are deps
  for styled-components v4, too, Users should already have them, as mentioned in the last PR
- Having all the new glitter (this does not really change anything, but adds a nice touch ; )

I squashed the commits for your convenience.

Edit: Nearly forgot: This PR closes #1

### CODE REVIEW

**Comment by morajabi** [comment]
@timhagn do you think you can resolve the conflicts and rebase? that'd be helpful! but if you can't no worries :)

**Comment by timhagn** [comment]
#done : )

File: package.json [hunk 1]
```diff
@@ -1,34 +1,54 @@
...
+    "dist",
+    "src"
+  ],
   "author": "Mohammad Rajabifard <mo.rajbi@gmail.com>",
   "license": "MIT",
   "peerDependencies": {
-    "styled-components": ">= 1 < 4"
+    "react": ">= 16.3.0",
+    "react-dom": ">= 16.3.0",
+    "styled-components": ">= 4"
```
Comment (morajabi) [inline package.json]: Looks like to me this branch is not against the master?

**Comment by morajabi** [comment]
@timhagn Tim, please add your name into AUTHORS file. Also, I see @ApacheEx isn't there too. That's meant to have everyone. 

Can you double check if this branch is rebased with current master? I see some unnecessary and outdated conflicts. For example the version range of styled-components doesn't match :)

File: package.json (hunk 1 above)
Comment (timhagn) [reply package.json]: Oh, had merged my latest master before the merge, but not afterwards.
Should be fixed now, unless you see more problems GitHub doesn't ; ).

**Comment by timhagn** [comment]
@morajabi Added myself to the AUTHORS file and updated the branch.

**Comment by timhagn** [comment]
Will close this for now and open another, less intrusive one : ).

**Pull request closed without merging**